src/Microsoft.DotNet.ProjectModel/CommonCompilerOptions.cs

```diff
@@ -39,7 +40,13 @@ public class CommonCompilerOptions
 
         public IEnumerable<string> AdditionalArguments { get; set; }
 
-        public string OutputName { get;set; }
+        public string OutputName { get; set; }
+
+        public string CompilerName { get; set; }
+
+        public IncludeFilesResolver EmbedInclude { get; set; }
```

Comment:
note that all the properties on `CommonCompilerOptions` are simply objects which are friendly to serialization. we should avoid attach property of complex object to it. @ajaybhargavb it's better to find different approach the carry this resolver around.